### PR TITLE
[Add] mrp_bom_reference_selection

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: mrp_bom_reference_selection/README.rst [hunk 1]
```diff
@@ -0,0 +1,56 @@
+MRP Bill of Material Reference Selection
+========================================
+
+This module was written to extend mrp features.
```
Comment (clonedagain) [inline mrp_bom_reference_selection/README.rst]: don't state the obvious...

File: mrp_bom_reference_selection/README.rst [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+Installation
+============
+
+To install this module, you just need to select the module and insure yourself dependancies are available.
+
+Configuration
+=============
+
+No particular configuration to use this module.
```
Comment (clonedagain) [inline mrp_bom_reference_selection/README.rst:16]: maybe remove this section then?

Comment (ghost) [inline mrp_bom_reference_selection/README.rst:16]: @clonedagain I agree :)

**Comment by ghost** [comment]
@foutoucour I would appreciate your review here.

File: mrp_bom_reference_selection/README.rst [hunk 3]
```diff
@@ -0,0 +1,55 @@
+MRP Bill of Material Reference Selection
+========================================
+
+This module allows to select the component in a bom when you have several bom for one product.
+This is used to manage versions of a product.
+Produced lot contain a reference to the bill of material used to compute the lot.
```
Comment (foutoucour) [inline mrp_bom_reference_selection/README.rst]: s,contain,contains

File: mrp_bom_reference_selection/__init__.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
```
Comment (foutoucour) [inline mrp_bom_reference_selection/__init__.py:22]: Could you explain why did you put some code in **init**.py. It is unusual at least.

It is important to note that odoo, when it starts, will read all the code in **init** files.

Could you just comment on your reasons?

File: mrp_bom_reference_selection/__init__.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+from . import models
+from . import wizards
+from openerp import SUPERUSER_ID
+
+
+def set_bill_of_material_references(cr, registry):
+    bom_obj = registry['mrp.bom']
```
Comment (foutoucour) [inline mrp_bom_reference_selection/__init__.py:34]: could you provide docstrings of this function?

**Comment by foutoucour** [comment]
Docstring and comments to add

then good to go :+1: 

(sorry for the delay :( )

File: mrp_bom_reference_selection/models/mrp_bom_line.py [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+    reference_id = fields.Many2one('mrp.bom.reference', 'Ref')
+    product_tmpl_id = fields.Many2one(
+        'product.template', 'Product Template',
+        related='product_id.product_tmpl_id', store=True)
+
+    product_id = fields.Many2one(
+        'product.product', required=True, string='Product')
+
+    @api.multi
+    def onchange_product_id(self, product_id):
```
Comment (bwrsandman) [inline mrp_bom_reference_selection/models/mrp_bom_line.py:38]: Why not use `@onchange`

File: mrp_bom_reference_selection/views/mrp_bom_view.xml [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+            </field>
+        </record>
+
+        <record id="mrp_bom_form_view" model="ir.ui.view">
+            <field name="name">mrp.bom.form</field>
+            <field name="model">mrp.bom</field>
+            <field name="inherit_id" ref="mrp.mrp_bom_form_view"/>
+            <field name="arch" type="xml">
+                <xpath expr="//field[@name='bom_line_ids']//field[@name='product_id']" position="replace">
+                    <field name="product_id" on_change="onchange_product_id(product_id)"/>
```
Comment (bwrsandman) [inline mrp_bom_reference_selection/views/mrp_bom_view.xml:23]: Why not use `@onchange` in the model

Comment (ghost) [inline mrp_bom_reference_selection/views/mrp_bom_view.xml:23]: Because @onchange does not support one2many relations

**Comment by bwrsandman** [comment]
:+1:

**Pull request merged**